src/app/components/share-button/share-button.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, Inject, Input } from '@angular/core';
+import { Clipboard } from '@angular/cdk/clipboard';
+import { Observable, tap } from 'rxjs';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { LoggerService } from '../../logger.service';
+import { first, map } from "rxjs/operators";
+import { DOCUMENT } from "@angular/common";
+
+@Component({
+  selector: 'app-share-button',
+  template: `
+    <button
+      mat-icon-button
+      matTooltip="URL to current example"
+      aria-label="URL to current example"
+      (click)="copyToClipboard()"
+    >
+      <mat-icon>share</mat-icon>
+    </button>
+  `,
+})
+export class ShareButtonComponent {
+  private readonly _name = 'ShareButtonComponent';
+
+  @Input()
+  url$!: Observable<string>;
+
+  constructor(
+    @Inject(DOCUMENT) private _document: Document,
+    private _clipboard: Clipboard,
+    private _snackBar: MatSnackBar,
+    private _logger: LoggerService,
+  ) { }
+
+  copyToClipboard() {
+    this.url$.pipe(
+      first(),
+      map((url) => `${this._document.location.origin}${url}`),
+      tap((url) => this._logger.logDebug(this._name, 'Copying URL to clipboard', { url })),
+      tap((url) => this._clipboard.copy(url)),
```

Comment:
What do you think about also replacing the current browser URL?

Reply:
It looks good to me. I can't remember the nuances of the router service w.r.t. this 😄